extensions/ql-vscode/src/query-history.ts

```diff
@@ -260,7 +266,10 @@ export class HistoryTreeDataProvider extends DisposableObject {
   }
 
   setCurrentItem(item?: QueryHistoryInfo) {
-    this.current = item;
+    if (item !== this.current) {
+      this.current = item;
+      this._onDidChangeCurrentQueryItem.fire(item);
```

Comment:
Looks like this is a bug that wasn't noticed before.

Reply:
It's a new event, and the previous code just had a potentially redundant field assignment. I don't think there was a real problem.